### ISSUE TITLE
Issue with TRE Upgrade to Version 0.17.0

### ISSUE THREAD

**Issue by radms** [microsoft/AzureTRE#3911]
When upgrading the TRE to the latest version, the upgrade is failing with the following error:

**Error:- Provider produced inconsistent final plan**

<img width="695" alt="image" src="https://github.com/microsoft/AzureTRE/assets/88878000/699d9058-b622-4efb-80b9-0357f137bd28">

**Comment by radms** [comment]
@tim-allen-ck No, the TRE was deployed and has been manually upgraded to the latest version.

To fix the above issue, the following changes were made to the api-webapp.tf file:
https://github.com/microsoft/AzureTRE/blob/main/core/terraform/api-webapp.tf)

<img width="596" alt="image" src="https://github.com/microsoft/AzureTRE/assets/88878000/e48dcd3f-d2f5-4090-9bb2-53aeaf0c1be4">

Once lines 61-63 were commented out, the code ran successfully.

**Comment by tim-allen-ck** [comment]
Hi @radms that's strange, I've built 0.17 quite a few times and not had to comment those lines out.
I have seen that error before though and usually a rerun of terraform fixes it.

**Comment by radms** [comment]
@tim-allen-ck We tried to rerun the code thrice, but we were still getting the same error on every execution. It only worked for us once we commented out the 3 lines.

**Comment by tim-allen-ck** [comment]
Could do with update the provider version for core

**Comment by tim-allen-ck** [comment]
Closing, Please reopen if the problem persists

**Comment by Danny-Cooke-CK** [comment]
@radms can you confirm the version of TRE you where on before the upgrade ?

**Comment by radms** [comment]
@Danny-Cooke-CK Before upgrade was using "v0.16.0".

**Comment by Danny-Cooke-CK** [comment]
For Clarity the underlying issue here was not with the Version 17 release but was actually due to a "make auth" command being run for a second time.  This updated the Application secret value's but it doesn't populate them to all necessary resources and hence the environment stopped working.

**Comment by radms** [comment]
@Danny-Cooke-CK  The issue with the 'make auth' command is separate from this one. We are encountering this particular issue during the deployment of the Terraform code.

**Comment by Danny-Cooke-CK** [comment]
Hi @radms. This ticket was escalated to us and when we spoke with you, you showed us the app secrets had been updated and you didn't understand why but it was causing the deployment to fail ? 
if thats a different issue then can you raise a different call please
